gateway-modules/apim-ballerina-configs/configs/org/wso2/carbon/apimgt/gateway/Authentication.bal

```diff
@@ -40,6 +40,17 @@ function authenticate (message m) (boolean, message) {
     string version = "1.0.0";
     string uriTemplate = "/*";
     string httpVerb = strings:toUpperCase(http:getMethod(m));
+    //check api status
+    string apiKey = apiContext + ":" + version;
+    dto:APIDTO apiDto = holder:getFromAPICache(apiKey);
+
+    if (constants:MAINTENANCE == apiDto.lifeCycleStatus) {
```

Comment:
Fixed in https://github.com/wso2/carbon-apimgt/pull/4087/commits/e9aae8173ce0104f16c45795dee53026a884e801